### ISSUE TITLE
Transforms are run for every provider, even if they are specified under a single provider

### ISSUE THREAD

**Issue by bfdykstra** [promptfoo/promptfoo#1756]
**Describe the bug**
I want to transform output from different providers differently, so I've written a provider transform

**To Reproduce**
Steps to reproduce the behavior, including example Promptfoo configurations if applicable:

promptfoo provider config:
```
providers:
  - id: "openai:chat:gpt-4o-mini"
    prompts:
      <...>
  - id: "anthropic:messages:claude-3-5-sonnet-20240620"
    transform: file://claude_transforms.py:special_claude_transform
    prompts:
      <...>
```

The transform:
```
def special_claude_transform(output, context):
    print(context['prompt']['label']) # <- will be run for any prompts defined under openai and for anthropic
    return context['prompt']['label'] + output
```

**Expected behavior**
I expect that a transform defined only under one provider will only be applied to prompts defined for that provider

**Screenshots**

**System information (please complete the following information):**

- Promptfoo version: `"promptfoo": "^0.89.4"`

**Additional context**

**Comment by mldangelo** [comment]
Thanks for reporting this @bfdykstra. It seems like a real bug, and I’ll look into it. And thank you for the configuration examples you provided. We’ll work on a fix soon. Thanks for helping improve Promptfoo!

**Comment by bfdykstra** [comment]
Thank you!

**Comment by mldangelo** [comment]
I tried reproducing this issue using a similar setup, but the transforms were only applied to the respective providers as expected. Here's the configuration I used:

```yaml
prompts:
  - id: 'Upper case: {{query}}'
    label: prompt1
  - id: 'prefix prompt {{query}}'
    label: prompt2
providers:
  - id: echo
    label: echo1
    transform: output.toUpperCase()
    prompts:
      - prompt1
  - id: echo
    label: echo2
    transform: '`prefix: ${output}`'
    prompts: 
      - prompt2
tests:
  - vars:
      query: injection
```

Could you please provide a bit more detail about your configuration or any additional context that might help us reproduce the behavior you're seeing? Thanks!

**Comment by bfdykstra** [comment]
hmm let me check again, may have been solved by #1757 ?

**Comment by bfdykstra** [comment]
Yup solved there!
